File: Locator/src/Vacancies/Vacancies.Contracts/Dto/GetVacanciesDto.cs
namespace Vacancies.Contracts.Dto;

public record GetVacanciesDto(
    string? SearchField,
    string? Experience,
    string? Employment,
    string? Schedule,
    int? Area,
    SalaryQuery? Salary,
    int? PerPage = null,
    int? Pages = null,
    int? Page = null);

public record SalaryQuery(long? Salary, string? Currency);

File: Locator/client/src/contracts/dtos/GetVacanciesDto.ts
export interface SalaryQuery {
    from?: number;
    to?: number;
    currency?: string;
}

export interface GetVacanciesDto {
    searchField?: string;
    experience?: string;
    employment?: string;
    schedule?: string;
    area?: string;
    salary?: SalaryQuery;
    perPage?: number;
    pages?: number;
    page?: number;
}

